tests/core/fs/workspacefs/test_workspacefile.py

```diff
@@ -0,0 +1,324 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import pytest
+import trio
+import io
+from parsec.core.fs.exceptions import FSUnsupportedOperation, FSOffsetError
+from parsec.core.types import FsPath
+
+
+@pytest.fixture
+@pytest.mark.trio
+async def trio_file(tmp_path):
+    d = tmp_path / "triofoo"
+    d.mkdir()
+    p = d / "triobar"
+    p.write_text("Hello World !")
+    return p
+
+
+@pytest.fixture
+@pytest.mark.trio
+async def random_text():
+    text = """Mem dem ima nequeam vos gratiam junctas aliunde maximam. Tot denuo terea tur
+    ritas justa talem vel. At possumus ac privatio superare in excitari tractant.
+    Advertisse incrementi quaerendum conservant denegassem ei in facillimum.
+    Ii mo utilius quamvis rationi ut fuerunt. Mea dignum vos ibidemtantae quidam cau quaeri cap.
+    Adipisci co de rationis ut originem competit sessione sequatur ad. Artificium frequenter agi
+    excoluisse mortalibus sum describere cau accidentia.
+    """
+    return text
+
+
+async def compare_read(triof=None, f=None, size: int = -1, alice_workspace=None, trio_file=None):
+    """ Comparing the full read of boths files with both methods"""
+    if (f is None and alice_workspace is None) or (triof is None and trio_file is None):
+        raise ValueError
+    if f is None:
+        f = await alice_workspace.open_file("/foo/bar", "rb")
+    if triof is None:
+        triof = await trio.open_file(trio_file, "rb")
+    output = await f.read(size)
+    output = output.decode("utf-8")
+    trio_output = await triof.read(size)
+    trio_output = trio_output.decode("utf-8")
```

Comment:
I don't think you need to decode the output of both files, comparing the raw output is fine